app/src/main/java/com/github/se/eventradar/viewmodel/qrCode/QrCodeAnalyser.kt

```diff
@@ -0,0 +1,59 @@
+package com.github.se.eventradar.viewmodel.qrCode
+
+import android.graphics.ImageFormat
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.BinaryBitmap
+import com.google.zxing.DecodeHintType
+import com.google.zxing.MultiFormatReader
+import com.google.zxing.PlanarYUVLuminanceSource
+import com.google.zxing.common.HybridBinarizer
+import java.nio.ByteBuffer
+
+class QrCodeAnalyser : ImageAnalysis.Analyzer {
+
+  // list of supported Image Formats
+  private val supportedImageFormats =
+      listOf(ImageFormat.YUV_420_888, ImageFormat.YUV_422_888, ImageFormat.YUV_444_888)
+
+  var onDecoded: ((String?) -> Unit)? = null //
+
+  override fun analyze(image: ImageProxy) {
+
+    // only want to scan if it is a QR Code
+    if (image.format in supportedImageFormats) {
+      val bytes = image.planes.first().buffer.toByteArray()
+
+      // parameters to scan
+      val source =
+          PlanarYUVLuminanceSource(
+              bytes, image.width, image.height, 0, 0, image.width, image.height, false)
+
+      val binaryBitmap = BinaryBitmap(HybridBinarizer(source))
+      try {
+
+        // result is the info encoded in to QR Code (String of userId in our case)
+        val result =
```

Comment:
One thing to think about as an extra (or possibly to write in PRD) is the security of this process, and its proneness to hacks and other attacks.

Reply:
Actually need to probably do a some sort of check: as you said if someone displays a random QR code they generated, that encodes a string... could be an attack.. need to think of some condition / property shared amongst all UIDs, could be length for e.g. if they all have a fixed length .... wouldn't eliminate attacks but defo cut it down a lot! should I do this or keep as extra?